src/api/app/policies/status/report_policy.rb

```diff
@@ -0,0 +1,46 @@
+class Status::ReportPolicy < ApplicationPolicy
+  def initialize(user, record)
+    raise Pundit::NotAuthorizedError, 'record does not exist' unless record
+    @user = user
+    @record = record
+  end
+
+  def create?
+    return false if @user.blank?
+    return true if @user.is_admin?
+    checkable_containers.all? { |container| write_permission_for_checkable?(container) }
+  end
+
+  def update?
+    create?
+  end
+
+  def destroy?
+    create?
+  end
+
+  def index?
+    true
+  end
+
+  def show?
+    index?
+  end
+
+  private
+
+  def checkable_containers
+    case @record.checkable
+    when BsRequest
+      @record.checkable.bs_request_actions.map(&:target_project_object)
+    when Repository
+      [@record.checkable.project]
+    end
+  end
+
+  def write_permission_for_checkable?(container)
+    container &&
+      (container.relationships.users.pluck(:user_id).include?(@user.id) ||
+      GroupsUser.where(group: container.relationships.groups.pluck(:group_id), user: @user).exists?)
```

Comment:
:thought_balloon: I really think this should be something generic in our objects that have user/group relationships. We keep doing this (`Object has user? || Object has a group with user`) in a gazillion places... But this is out of scope for this PR I guess...